src/test/java/uk/gov/justice/digital/hmpps/keyworker/events/ComplexityOfNeedEventProcessorTest.kt

```diff
@@ -38,25 +47,46 @@ class ComplexityOfNeedEventProcessorTest {
   @Test
   fun `should not deallocate offenders that do not have high complexity of needs`() {
     complexityOfNeedEventProcessor =
-      ComplexityOfNeedEventProcessor(keyworkerService, telemetryClient, objectMapper, "http://local")
+      ComplexityOfNeedEventProcessor(
+        keyworkerService,
+        telemetryClient,
+        objectMapper,
+        "http://local",
+        complexityOfNeedGateway,
+        allocationContextHolder,
+      )
     complexityOfNeedEventProcessor.onComplexityChange(cOMPLEXITYMESSAGELOW)
 
-    verify(keyworkerService, never()).deallocate(OFFENDER_NO)
+    verify(keyworkerService, never()).deallocate(OFFENDER_NO, CHANGE_IN_COMPLEXITY_OF_NEED)
   }
 
   @Test
   fun `should deallocate offenders that have high complexity of needs`() {
     complexityOfNeedEventProcessor =
-      ComplexityOfNeedEventProcessor(keyworkerService, telemetryClient, objectMapper, "http://local")
+      ComplexityOfNeedEventProcessor(
+        keyworkerService,
+        telemetryClient,
+        objectMapper,
+        "http://local",
+        complexityOfNeedGateway,
+        allocationContextHolder,
+      )
     complexityOfNeedEventProcessor.onComplexityChange(cOMPLEXITYMESSAGEHIGH)
 
-    verify(keyworkerService, times(1)).deallocate(OFFENDER_NO)
+    verify(keyworkerService, times(1)).deallocate(OFFENDER_NO, CHANGE_IN_COMPLEXITY_OF_NEED)
```

Comment:
Might be good to test that the username from the CoN API response is used as the deallocate by